Sources/StreamChat/Utils/LazyCachedMapCollection.swift

```diff
@@ -5,98 +5,60 @@
     public init(_ collection: LazyCachedMapCollection) {
-        // Reuse the generator
-        generator = collection.generator
-        // Create a new cache instance
-        cache = .init(capacity: collection.count)
-        // After creating the new cache, refill the existing cached values
-        cache.values = collection.cache.values
+        mappedElements = collection.mappedElements
+    }
+    
+    private let mappedElements: [Element]
+    
+    init(elements: [Element]) {
+        mappedElements = elements
     }
 
     public init<Collection: RandomAccessCollection, SourceElement>(
         source: Collection,
         map: @escaping (SourceElement) -> Element,
         context: NSManagedObjectContext? = nil
     ) where Collection.Element == SourceElement, Collection.Index == Index {
-        // In v5 we should deprecate LazyCachedMapCollection since it is very error prone
-        // And customers should not have access to these helper data structures
-        // This is not great, but we need to make sure that when mapping DTOs to models
-        // that these DTOs are accessed on their managed object contexts.
-        // So we need to pass the NSManagedObjectContext to the generator.
-        generator = { index in
-            var element: Element!
-            if let context = context {
-                context.performAndWait {
-                    element = map(source[index])
-                }
-            } else {
-                element = map(source[index])
-            }
-            return element
-        }
-
-        /// This is just an internal test to see how we behave when the DB models are immediately mapped instead of being lazily mapped
-        if !StreamRuntimeCheck._isBackgroundMappingEnabled {
-            cache = .init(capacity: source.count)
-        } else {
-            if Thread.isMainThread {
-                log.debug("This should not be happening on the Main Thread")
-            }
-            cache = .init(elements: source.map(map))
-        }
+        mappedElements = source.map(map)
     }
 
-    internal var generator: (Index) -> Element
-    internal var cache: Cache<Element>
-
     public subscript(position: Index) -> Element {
-        if let cached = cache.values[position] {
-            return cached
-        } else {
-            let value = generator(position)
-            defer { cache.values[position] = value }
-            return value
-        }
+        mappedElements[position]
     }
 
     public mutating func append(_ element: Element) {
-        // Even though `LazyCachedMapCollection` is a value type,
-        // the `cache` is not, so whenever we append a new value,
-        // we need to make sure we are creating a copy.
-        self = LazyCachedMapCollection(self)
-        cache.values.append(element)
+        var elements = mappedElements
+        elements.append(element)
+        self = LazyCachedMapCollection(elements: elements)
```

Comment:
Can we just do this? 🤔 
```suggestion
        mappedElements.append(element)
```

Reply:
Yes, if I make it var. Changed it